### DISCUSSION TITLE
Review of AI

### DISCUSSION THREAD

**Discussion by AlexeyMatveev686** [ONLYOFFICE/onlyoffice.github.io#248] [Polls]
<img src="https://raw.githubusercontent.com/ONLYOFFICE/onlyoffice.github.io/master/discussions-icons/openai.svg">

**Comment by Markoise** [comment]


**Reply by AlexeyMatveev686** [reply]
Hello, this plugin works through context menu into the document.
![chatGPT](https://github.com/ONLYOFFICE/onlyoffice.github.io/assets/34643750/c826a24d-ae83-4390-b240-7d5563239210)

**Reply by Loweel** [reply]
It just does not work. Whatever I click, nothing is changed. I downloaded the last edition of OnlyOffice. I put the key. I have a subscription. Still, nothing works. Context menu exists,but id does notthing. There nothing more in the setting I can change. Besides, it only uses davinci-003 , while I have subscription for chatgpt4

**Reply by AlexeyMatveev686** [reply]
We have updated this plugin to version 1.1.4. Now it can work with gpt-4 model if you have a subscription. Please update it and try to use again.

**Comment by pigeonskiller** [comment]
It does not work. Tested both the latest desktop version and the previous one on Linux and Windows with different API keys. Both give the same error even though they have not exceeded any quota:
"Error:
You exceeded your current quota, please check your plan and billing details. For more information on this error, read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors."
The same error is reported by many other users.
It probably only works with a paid API, but OnlyOffice should declare it...

**Reply by AlexeyMatveev686** [reply]
Hello. Yes, it works only with a paid API. We will add it into Readme. Thanks for your notification.

**Reply by pigeonskiller** [reply]
You should also consider free alternatives since OnlyOffice desktop is free – it would be pointless otherwise!
I suggest you consider HuggingChat at the link https://huggingface.co/chat/: it works reasonably well, contains several models (even better than Llama 2 70B) and perform web searches!
In case you need a beta tester for this eventual plugin I can offer my help for free.
Greetings.

**Reply by Carlos-err406** [reply]
also would be nice to allow openai-compatible apis
instead of hardcoding the openai url and models

![image](https://github.com/ONLYOFFICE/onlyoffice.github.io/assets/81443707/1556a6bd-14de-4aac-bb09-20430780ade6)

**Comment by AlexeyMatveev686** [comment]
Thanks for your offer. We will check this service. If it's possible to make plugin with it, we will add it into a list of the feature plugins.

**Comment by kenneth-mid** [comment]
A great plugin, it can only be used in Word right now, right?
I'm very sorry that I made a Chinese language pack without permission to avoid misoperation during use.
I feel that the current functions are still relatively simple, and when I choose to optimize the grammar, it deletes all the content. I don't know why yet. I am using GPT4.

**Reply by AlexeyMatveev686** [reply]
Hello. Full functionality works only into a document editor. But you can make a custom request and use a chat into a slide editor and cell editor. Unfortunally now we don't have another method for replace a text into a document. If you have some problem with text replacing, please create an issue about it.

**Comment by kenneth-mid** [comment]
There's another idea, I don't know if it can be implemented, because I don't know how to handle this problem due to insufficient programming development skills.
Can the API interface settings in the plugin be made into parameter options? Like the key code, this would make it easier for us to choose other interface methods when using it

**Reply by AlexeyMatveev686** [reply]
Do you mean to add some additional settings for the model to the settings window? How is it in the custom request window?

**Comment by yannrichet-asnr** [comment]
Great plugin, works fine in desktop, but not working in docs as far as I tried.

It should be greater if **completion** could be also available while typing in a doc/cell/...
=> Definitively the best usage IMO.

**Reply by Misokados** [reply]
Hello @yannrichet-asnr 
Please describe the situation you faced. We need step-by-step usage scenario to check it out.

**Reply by yannrichet-asnr** [reply]
Now working on 9.2.
But still waiting for an AI-completion engine (almost like any code IDE) based on all opened documents used as RAG source/context.

**Reply by Misokados** [reply]
We’re currently working on supporting RAG (internal tracknumber - 69928). We will contact you once we have something to share.

**Comment by dipping5115** [comment]
Dear developers, could you please differentiate between simplified Chinese and traditional Chinese in the translation function, as the differences between these two languages are quite significant.

**Reply by Misokados** [reply]
Hello @dipping5115 
We're checking your request, I will contact you shortly.

**Reply by Misokados** [reply]
Dear @dipping5115 
We have added your request (76683) to internal tracksystem, and we have started working on it. 
We will contact you once we have something to share.

**Comment by wintsworks** [comment]
Doesn't work with Gemini, also doesn't work with remote Ollama.

**Reply by xxLeoxx93** [reply]
yep, still doesnt 6 months later...

**Reply by Misokados** [reply]
Hello,
Gemini and Ollama are supported AI models. Please describe the situation step-by-step, we will check it out. To avoid confusion in this thread, you can create a separate post on https://community.onlyoffice.com/ and describe the issue you faced.

**Comment by Atom02** [comment]
how to make it work open router?

**Reply by Misokados** [reply]
Hello,
please describe the situation as detailed as possible. We will check it out.

**Comment by vovinhd** [comment]
Hey, quick question: I didn't install this. How do I remove it?

**Reply by WhileTrue-Continue** [reply]
Same.  I'd like to get rid of all these AI add ons, but it's weird that they are being made the most difficult bits to remove

**Reply by Misokados** [reply]
Hello, the ability to remove the AI plugin has been added and currently available. I’ve checked  v.9.2.1 for the desktop app and the self-hosted Document Server.

**Comment by regs01** [comment]
Ollama is not working. Can't retrieve models. Ollama is set to IP, not to localhost. Chatbox, Continue and WebUI are working.

**Reply by Misokados** [reply]
Hello,
Ollama is supported AI model. Please describe the situation step-by-step, we will check it out. To avoid confusion in this thread, you can create a separate post on https://community.onlyoffice.com/ and describe the issue you faced.

**Reply by regs01** [reply]
> Hello, Ollama is supported AI model. Please describe the situation step-by-step, we will check it out. To avoid confusion in this thread, you can create a separate post on https://community.onlyoffice.com/ and describe the issue you faced.

It just doesn't work out of box. As far as I remember needs OLLAMA_ORIGINS=* environmental variable to be set. So needs to be better documented.

**Reply by Misokados** [reply]
Yeas, some setup steps are required. We constantly improve the AI plugin and I expect that it will become easier to set it up in the future. meanwhile, we posted detailed guide here: https://helpcenter.onlyoffice.com/docs/userguides/ai/ollama.aspx

**Reply by regs01** [reply]
That's macOS only. Not Windows.

**Reply by Misokados** [reply]
We're going to update the guide shortly and provide the steps for Windows.

**Reply by erenirmak** [reply]
after 5 months, still not updated for windows. and ollama is not working for me.
(remote ollama on windows, documentserver container on raspberry pi, on the same LAN)

**Reply by Misokados** [reply]
Hello @erenirmak 
The guide is available here: https://api.onlyoffice.com/docs/ai/guides/configuring-ollama-with-cors/

**Comment by Sollace** [comment]
This plugin was added in the background without my consent. 👎

**Reply by regs01** [reply]
> This plugin was added in the background without my consent. 👎


It's not working unless you set it up.

**Reply by Sollace** [reply]
> > This plugin was added in the background without my consent. 👎
> 
> It's not working unless you set it up.

1. It still adds an AI tab to the interface
2. Working or not, I didn't install it. It shouldn't be here.

Also there's no way to fully uninstall it. I can turn it off but it's still there. I want it deleted off my computer. How do I do that?

**Reply by regs01** [reply]
> > > This plugin was added in the background without my consent. 👎
> > 
> > 
> > It's not working unless you set it up.
> 
>     1. It still adds an AI tab to the interface
> 
>     2. Working or not, I didn't install it. It shouldn't be here.
> 
> 
> Also there's no way to fully uninstall it. I can turn it off but it's still there. I want it deleted off my computer. How do I do that?

It's a stock plug in. It doesn't install on its own, it comes with OnlyOffice as several more stock plugins as well.

**Reply by Sollace** [reply]
At least let us uninstall it then

**Reply by Sollace** [reply]
Youtube and Translate came preinstalled and they can be removed. Why not the AI plugin?
<img width="1108" height="729" alt="image" src="https://github.com/user-attachments/assets/c3f165d0-6048-4c48-92ab-d5238c687a7e" />

**Reply by Sollace** [reply]
That said, there seem to be a whole bunch of plugins that cannot be removed or disabled.
<img width="3064" height="1871" alt="image" src="https://github.com/user-attachments/assets/2d577543-a0f6-4d53-9c3c-43ec543e95c7" />

**Reply by regs01** [reply]
They are stock functionality made in form of plugins, as an example.

**Reply by Misokados** [reply]
Hello, the ability to remove the AI plugin has been added and currently available. I’ve checked v.9.2.1 for the desktop app and the self-hosted Document Server.

**Reply by LexGear** [reply]
Thank goodness, because in my current version I cannot remove it.
<img width="584" height="410" alt="image" src="https://github.com/user-attachments/assets/4f36ee80-6f06-4a02-8ddf-72343adf1f06" />

**Reply by Misokados** [reply]
Please accept my sincere apologies for misunderstanding. We're still working to add the requested ability. Meanwhile, please refer to the following post for instructions on how to remove the plugin manually: https://github.com/ONLYOFFICE/onlyoffice.github.io/discussions/248#discussioncomment-15933396

**Comment by vn27071979-ux** [comment]
Вообще, ничего не работает. Ни один, ИИ ассистент! Вот это да, молодцы!

**Reply by regs01** [reply]
Так а ты их подключил? Большая часть платная. А Олламу работает, но настроить нужно. Но уже не полностью что. Вроде глобальной переменной какой-то.

**Reply by Misokados** [reply]
Hello, 
Please describe the situation you faced. We need step-by-step usage scenario to check it out.

**Comment by regs01** [comment]
There is a problem with many models with Ollama. When translating multiple cells horizontally results from all cells are inserted into first cell. Particularly `Hunyuan MT`, `Mistral-nemo` etc. With some models problem also exist vertically.

**Reply by Misokados** [reply]
Hello,
Please describe the situation step-by-step. We will check it out. You can create a video demonstration or make screenshots. To avoid confusion in this thread, you can create a separate post on https://community.onlyoffice.com/ and describe the issue you faced.

**Comment by xxLeoxx93** [comment]
Seems like it trys to use wrong endpoints (LM Studio over Network, Error in LM Studio: Unexpected Endpoint or method, returning 200 anyway). Therefore can't be used.

**Reply by Misokados** [reply]
Please describe the situation step-by-step. We will check it out. You can create a video demonstration or make screenshots. To avoid confusion in this thread, you can create a separate post on https://community.onlyoffice.com/ and describe the issue you faced.

**Comment by mehgan792-cpu** [comment]
how to remove this 'plugin'?

**Reply by regs01** [reply]
It's a part of office. You just disable AI providers (and they are disabled by default) and it will be disabled.

**Reply by Misokados** [reply]
Hello, the ability to remove the AI plugin has been added and currently available. I’ve checked v.9.2.1 for the desktop app and the self-hosted Document Server.

**Reply by ZakariaHassine** [reply]
> Hello, the ability to remove the AI plugin has been added and currently available. I’ve checked v.9.2.1 for the desktop app and the self-hosted Document Server.

I have version 9.2.1 and I can't remove it. The "Remove" button, if exists, does only remove the plugin update (if installed) and not the plugin itself.

<img width="1920" height="1038" alt="Screenshot From 2026-02-13 21-31-34" src="https://github.com/user-attachments/assets/a6b2c3a8-6e86-4f4b-8270-9a5cea3f9145" />

Notice that I can't remove, at least, the first 9 plugins:

<img width="605" height="990" alt="Screenshot From 2026-02-13 21-40-32" src="https://github.com/user-attachments/assets/014d9606-21e1-4e95-9bde-a26a8dd8eefa" />

**Reply by Misokados** [reply]
Please update the plugin and re-check the situation:
<img width="606" height="617" alt="image" src="https://github.com/user-attachments/assets/c7b17ef8-5f04-4f71-be5b-390c72c5def2" />
Please let us know the result.

**Reply by ZakariaHassine** [reply]
Before updating the plugin:
<img width="616" height="642" alt="Screenshot From 2026-02-18 20-34-03" src="https://github.com/user-attachments/assets/28360a1b-17d0-4799-afa6-7e128615dfb8" />
After updating the plugin:
<img width="616" height="642" alt="Screenshot From 2026-02-18 20-34-22" src="https://github.com/user-attachments/assets/b2885d7a-1b54-4fc6-a06b-2576382aa2e4" />
After closing and re-opening the Plugin Manager:
<img width="616" height="642" alt="Screenshot From 2026-02-18 20-34-40" src="https://github.com/user-attachments/assets/1feb1086-96d1-4da8-a0f7-d081fb9ad528" />
Note: that "Update" button won't disappear after updating again and closing & re-opening the Plugin Manager or after restarting ONLYOFFICE app itself. I tried all different scenarios.
After removing the plugin (unfortunately, it's too soon to celebrate):
<img width="616" height="642" alt="Screenshot From 2026-02-18 20-34-55" src="https://github.com/user-attachments/assets/657635a4-4927-478c-a9af-07f4a51ac2af" />
After closing & re-opening the Plugin Manager:
<img width="616" height="642" alt="Screenshot From 2026-02-18 20-35-12" src="https://github.com/user-attachments/assets/e224a336-1916-4d62-8c8d-06f3ae94adda" />
After restarting ONLYOFFICE:
<img width="616" height="642" alt="Screenshot From 2026-02-18 20-36-10" src="https://github.com/user-attachments/assets/55123d51-1691-4703-bb97-ea40c4cbdac5" />

**Reply by Misokados** [reply]
Hello @ZakariaHassine 
Thank you, we're checking the situation.

**Reply by Misokados** [reply]
Thank you for the provided data!
Please accept my sincere apologies for misunderstanding. At the moment there's no way to remove some plugins via Plugin manager, and we're working on it (bugs: 70265, 76364).
The workaround solution for AI plugin is removing specific folder as described here: https://community.onlyoffice.com/t/how-can-i-remove-ai/17202/4
NB! These actions are performed at your own risk. We cannot guarantee stable operation after making these changes.

**Reply by ZakariaHassine** [reply]
No issues.

For flatpak version, plugins to remove are located in ```/var/lib/flatpak/app/org.onlyoffice.desktopeditors/current/active/files/bin/opt/onlyoffice/desktopeditors/editors/sdkjs-plugins```, but a future update might likely restore/reinstall the plugins again. Plugins needs to be moved to ```~/.var/app/org.onlyoffice.desktopeditors/data/onlyoffice/desktopeditors/sdkjs-plugins```.

A ```systemd service``` or a ```cron job scheduler```, might do the job, but not everyone is familiar with this kind of automation.

Thanks for the link.

**Comment by vpulido-creator** [comment]
Should get a more robust Anthropic integration (or other provider) - falls short versus integration in other products

**Reply by Misokados** [reply]
Hello @vpulido-creator 
We're constantly improving the AI plugin, so we really appreciate your feedback. Could you please share a bit more detail about what you would like to see in the Anthropic integration? We'll check it out.

**Reply by regs01** [reply]
I'd say better integration with free models with Ollama. As some translation models requires specific instructions and have specific answer format. So when you translate multiple cells results won't insert back all in one cell.

**Comment by QBox-GHub** [comment]
Hi there,
As much as I would love to use this plugin with ollama on a remote machine, it doesn't seem to work. I tried to:

1. enable/disable plugin
2. remove / reinstall
3. restart onlyoffice
4. connect to remote ollama via web and it works fine
5. manually enter URL again using hostname, IP Address, FQDN. (none seem to work or make any change to _**Failed to Fetch**_)

I'm on latest version of ONLYOFFICE 9.3.1.8 (x64 deb) on fully updated Ubuntu 25.10
At the moment I tested with Ollama only.

Any tips would be appreciated

Thanks

**Reply by regs01** [reply]
As far as I remember you need OLLAMA_ORIGINS=* environmental variable, which is not safe, if Ollama is open externaly.

**Reply by QBox-GHub** [reply]
> As far as I remember you need OLLAMA_ORIGINS=* environmental variable, which is not safe, if Ollama is open externaly.

Thanks @regs01 it worked like a champ! I might have missed this piece of info in the plugin instructions!
BTW, I'm just testing Ollama on Proxmox VM where I'm running models locally. Once I check a few models I will harden access via strict firewall rules as per the rest of the lab! Thanks again for your help.
